tests/callbacks/test_stochastic_weight_avg.py

```diff
@@ -63,22 +80,41 @@ def configure_optimizers(self):
             },
         }
 
+    def training_epoch_end(self, _):
+        if not self.crash_after_epoch:
+            return
+        self._epoch_count += 1
+        if self._epoch_count >= self.crash_after_epoch:
+            raise RuntimeError("Crash test")
+
 
 class SwaTestCallback(StochasticWeightAveraging):
+    def __init__(self, *args, **kwargs):
+        if "resuming_from_epoch" in kwargs:
+            self.resuming_from_epoch = kwargs["resuming_from_epoch"]
+            del kwargs["resuming_from_epoch"]
+        else:
+            self.resuming_from_epoch = 0
+        super().__init__(*args, **kwargs)
+
+    validation_calls: int = 0
     update_parameters_calls: int = 0
     transfer_weights_calls: int = 0
 
     def update_parameters(self, *args, **kwargs):
         self.update_parameters_calls += 1
         return StochasticWeightAveraging.update_parameters(*args, **kwargs)
 
+    def on_validation_start(self, *args, **kwargs):
+        self.validation_calls += 1
```

Comment:
This doesn't check that the averaged model was used in validation, right? Could you maybe add a test for this?

Reply:
Yes I'll add this in the follow up PR for validation now that this is reverted.